x-pack/plugins/ml/common/util/object_utils.ts

```diff
@@ -34,3 +34,14 @@ export const isPopulatedObject = <U extends string = string>(
       requiredAttributes.every((d) => ({}.hasOwnProperty.call(arg, d))))
   );
 };
+
+/**
+ * Get the first key in the object
+ * getFirstKeyInObject({ firstKey: {}, secondKey: {}}) -> firstKey
+ */
+export const getFirstKeyInObject = (arg: unknown): string | undefined => {
+  if (isPopulatedObject(arg)) {
+    const keys = Object.keys(arg);
+    return keys.length > 0 ? keys[0] : undefined;
+  }
+};
```

Comment:
question: what if the object key is `undefined`, e.g.   
```typescript
const myObj = { undefined: 'my-value' }
```
how can we distinguish the result?

Reply:
I think the `undefined` in myObj will be coerced into a string according to the [specification](https://262.ecma-international.org/5.1/#sec-4.2.1). In which case I think it's correct that `getFirstKeyInObject(myObj) => 'undefined'`.